ddl/split_region.go

```diff
@@ -70,9 +69,17 @@ func splitPreSplitedTable(store kv.SplitableStore, tbInfo *model.TableInfo, scat
 	// And the max _tidb_rowid is 9223372036854775807, it won't be negative number.
 
 	// Split table region.
-	step := int64(1 << (tbInfo.ShardRowIDBits - tbInfo.PreSplitRegions))
+	preSplitRegionCount := tbInfo.PreSplitRegionCount
+	if preSplitRegionCount == 0 {
+		preSplitRegionCount = 1 << tbInfo.PreSplitRegions
+	}
+
+	step := int64((1 << tbInfo.ShardRowIDBits) / preSplitRegionCount)
+	if step == 0 {
```

Comment:
I think we'd better unify this check. Could we put it to [line1805](https://github.com/pingcap/tidb/pull/12688/files?file-filters%5B%5D=.go#diff-edfa54cbe2fdb418f0a77bd0ec4a56f2R1805)?